modules/05_add_blob_storage/src/api.py

```diff
@@ -0,0 +1,88 @@
+from flask import Flask, flash, render_template, request, redirect, session, Response
+from markupsafe import Markup
+
+from .in_memory_storage import InMemoryStorage, StorageItem
+from .blob_storage import BlobStorage
+
+APP_VERSION = '0.0.1'
+
+app = Flask(__name__)
+app.secret_key = "f3cfe9ed8fae309f02079dbf"
+
+database = InMemoryStorage()
+image_storage : BlobStorage = BlobStorage()
+
+@app.context_processor
+def inject_app_version():
+    return dict(app_version=APP_VERSION)
+
+
+@app.route('/')
+def home():
+    return render_template('home.html')
+
+
+@app.route('/game', methods=['GET'])
+def game():
+    if database.is_empty():
+        flash("No images uploaded yet! Please upload at least one image to start guessing")
+        return redirect("/")
+
+    if ('secret_item_id' not in session or
+        not database.has_index(session['secret_item_id'])
+    ):
+        # need to renew 'secret_item_id' in the session.
+        # it's either missing or left over from the old version of the app
+        session['secret_item_id'] = database.get_random_item_index()
+
+    return render_template('game.html')  # continue the game
+
+
+@app.route('/current_image', methods=['GET'])
+def current_image():
+    secret_item = database.get_item_by_index(session['secret_item_id'])
+    return Response(image_storage.download_image(secret_item.image_name), mimetype=secret_item.image_content_type)
```

Comment:
Either `return redirect(image_url, code=302)` or, more preferrably, insert direct image url right in the  jinja template and remove this endpoint altogether

Reply:
I kept the redirect, it looks clearer to me.

Reply:
Alright! But please get the latest version of templates/this endpoint in 04_add_images, I changed it